webhook/server/server.go

```diff
@@ -125,36 +125,14 @@ func (m *podInjector) Handle(ctx context.Context, req admission.Request) admissi
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	logger.Info("injecting into Pod", "name", pod.Name, "generatedName", pod.GenerateName, "namespace", req.Namespace)
+	logger.Info("checking pod", "name", pod.Name, "generatedName", pod.GenerateName, "namespace", req.Namespace)
 
-	var ns corev1.Namespace
-	if err := m.client.Get(ctx, client.ObjectKey{Name: req.Namespace}, &ns); err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
-
-	inject := utils.GetField(ns.Annotations, dtwebhook.AnnotationInject, "true")
-	inject = utils.GetField(pod.Annotations, dtwebhook.AnnotationInject, inject)
-	if inject == "false" {
-		return admission.Patched("")
-	}
-
-	oaName := utils.GetField(ns.Labels, dtwebhook.LabelInstance, "")
-	if oaName == "" {
-		return admission.Errored(http.StatusBadRequest, fmt.Errorf("no DynaKube instance set for namespace: %s", req.Namespace))
+	oa, response := m.FindDynakubeForPod(ctx, pod)
```

Comment:
Note that this high-throughput function. It's currently O(\<number of DynaKubes\>*\<number of Pods\>). Whether how relevant that is needs to be calculated.

There are alternatives, such as the Namespace controller maintaining a ConfigMap with the mappings from namespace to DynaKube that can be queried once.

Reply:
@baichinger, what is an example of a conflicting configuration? If you mean finding two dynakubes that are responsible for a single namespace, it already prints an error accordingly and does not inject